saleor/webhook/tests/test_webhook_payloads.py

```diff
@@ -82,3 +90,62 @@ def test_order_lines_have_all_required_fields(order, order_line_with_one_allocat
         ),
         "tax_rate": str(line.tax_rate.quantize(Decimal("0.0001"))),
     }
+
+
+def test_generate_product_variant_payload(
+    product_with_variant_with_two_attributes, product_with_images
+):
+    variant = product_with_variant_with_two_attributes.variants.first()
+    payload = json.loads(generate_product_variant_payload(variant))[0]
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.id)
+    additional_fields = ["channel_listings"]
+    extra_dict_data = ["attributes", "product_id", "images"]
+    payload_fields = list(
+        chain(
+            ["id", "type"], PRODUCT_VARIANT_FIELDS, extra_dict_data, additional_fields
+        )
+    )
+
+    for field in payload_fields:
+        assert payload.get(field) is not None
+
+    assert variant_id is not None
+    assert payload["sku"] == "prodVar1"
+    assert len(payload["attributes"]) == 2
+    assert len(payload["channel_listings"]) == 1
+    assert payload["channel_listings"][0] == {
+        "cost_price_amount": "1.000",
+        "currency": "USD",
+        "id": ANY,
+        "price_amount": "10.000",
+        "type": "ProductVariantChannelListing",
+    }
+    assert len(payload.keys()) == len(payload_fields)
+
+
+def test_generate_product_variant_deleted_payload(
+    product_with_variant_with_two_attributes,
+):
+    variant = product_with_variant_with_two_attributes.variants.prefetch_related(
```

Comment:
IMO to proper simulate the case we should call .delete() in the same way as we did in mutation responsible for  `.delete` action.
@fowczarek  What do you think? 
Here we already fetched the object, so even if we call delete on QS, we will have all details stored locally. It is little bit different way of deleting than we currently use

Reply:
In mutation it's handled the sam way. Object/s is/r being cached before actually being removed, and then
 passed to the event.